ibm/service/power/resource_ibm_pi_network_security_group_rule.go

```diff
@@ -0,0 +1,654 @@
+			mbrMap := networkSecurityGroupMemberToMap(mbr)
+			members = append(members, mbrMap)
+		}
+		d.Set(Attr_Members, members)
+	} else {
+		d.Set(Attr_Members, nil)
+	}
+
+	d.Set(Attr_NetworkSecurityGroupID, networkSecurityGroup.ID)
+	if len(networkSecurityGroup.Rules) > 0 {
+		rules := []map[string]interface{}{}
+		for _, rule := range networkSecurityGroup.Rules {
+			ruleMap := networkSecurityGroupRuleToMap(rule)
+			rules = append(rules, ruleMap)
+		}
+		d.Set(Attr_Rules, rules)
+	}
+	if len(Attr_UserTags) > 0 {
+		d.Set(Attr_UserTags, networkSecurityGroup.UserTags)
+	}
+	return nil
+}
+
+func resourceIBMPINetworkSecurityGroupRuleDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	ids, err := flex.IdParts(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if len(ids) == 3 {
+		cloudInstanceID := ids[0]
+		nsgID := ids[1]
+		ruleID := ids[2]
+
+		sess, err := meta.(conns.ClientSession).IBMPISession()
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		nsgClient := instance.NewIBMIPINetworkSecurityGroupClient(ctx, sess, cloudInstanceID)
+
+		err = nsgClient.DeleteRule(nsgID, ruleID)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		_, err = isWaitForIBMPINetworkSecurityGroupRuleRemove(ctx, nsgClient, nsgID, ruleID, d.Timeout(schema.TimeoutDelete))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+	d.SetId("")
+	return nil
+}
+
+func isWaitForIBMPINetworkSecurityGroupRuleAdd(ctx context.Context, client *instance.IBMPINetworkSecurityGroupClient, id, ruleID string, timeout time.Duration) (interface{}, error) {
+
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_Pending},
+		Target:     []string{State_Available},
+		Refresh:    isIBMPINetworkSecurityGroupRuleAddRefreshFunc(client, id, ruleID),
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Minute,
+	}
+
+	return stateConf.WaitForStateContext(ctx)
+}
+
+func isIBMPINetworkSecurityGroupRuleAddRefreshFunc(client *instance.IBMPINetworkSecurityGroupClient, id, ruleID string) retry.StateRefreshFunc {
+
+	return func() (interface{}, string, error) {
+		networkSecurityGroup, err := client.Get(id)
+		if err != nil {
+			return nil, "", err
+		}
+
+		if networkSecurityGroup.Rules != nil {
+			for _, rule := range networkSecurityGroup.Rules {
+				if *rule.ID == ruleID {
+					return networkSecurityGroup, State_Available, nil
+				}
+
+			}
+		}
+		return networkSecurityGroup, State_Pending, nil
+	}
+}
+
+func isWaitForIBMPINetworkSecurityGroupRuleRemove(ctx context.Context, client *instance.IBMPINetworkSecurityGroupClient, id, ruleID string, timeout time.Duration) (interface{}, error) {
+
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_Pending},
+		Target:     []string{State_Removed},
+		Refresh:    isIBMPINetworkSecurityGroupRuleRemoveRefreshFunc(client, id, ruleID),
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Minute,
```

Comment:
```suggestion
		MinTimeout: 10 * time.Minute,
```
This can be reduce to 60 sec

Reply:
updated